packages/desktop-client/src/components/budget/MobileBudgetTable.js

```diff
@@ -774,10 +850,25 @@ class BudgetGroups extends Component {
 
 export function BudgetTable(props) {
   const [editingCategory, setEditingCategory] = useState(null);
-
+  const [smallVPWidth, setSmallVPWidth] = useState(window.innerWidth < 365);
   function onEditCategory(id) {
     setEditingCategory(id);
   }
+  useEffect(() => {
+    const handleResize = () => {
+      if (window.innerWidth < 365) {
+        setSmallVPWidth(true);
+      } else {
+        setSmallVPWidth(false);
+      }
+    };
+
+    window.addEventListener('resize', handleResize);
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏this is a great start. But could you move it to the highest level component in this file and then just pass down as a prop in order to reduce the duplicated logic?

Thanks